onmt/models/model.py

```diff
@@ -42,7 +43,10 @@ def forward(self, src, tgt, lengths, dec_state=None):
         """
         tgt = tgt[:-1]  # exclude last target from inputs
 
-        enc_final, memory_bank = self.encoder(src, lengths)
+        if self.model_type == 'audio':
+            enc_final, memory_bank, lengths = self.encoder(src, lengths)
```

Comment:
instead of making a specific case, why don't make all encoder interfaces the same?